test/unit/create-template-orders.spec.js

```diff
@@ -0,0 +1,498 @@
+        .reply(200)
+
+      const setIsSubscriptionProcessed = nock(CTP_API_URL)
+        .post(
+          `/${PROJECT_KEY}/orders/${CHECKOUT_ORDER_ID}`,
+          (body) =>
+            JSON.stringify(body) ===
+            JSON.stringify({
+              actions: [
+                {
+                  action: 'setCustomField',
+                  name: 'isSubscriptionProcessed',
+                  value: true,
+                },
+              ],
+              version: 3,
+            })
+        )
+        .reply(200)
+
+      const updateLastStartTimestamp = nock(CTP_API_URL)
+        .post(
+          `/${PROJECT_KEY}/custom-objects`,
+          (body) =>
+            body.container === 'commercetools-subscriptions' &&
+            body.key === 'subscriptions-lastStartTimestamp'
+        )
+        .reply(200)
+
+      const stats = await createTemplateOrders({
+        ctpClient,
+        apiRoot,
+        logger,
+        startDate: new Date(),
+      })
+
+      expect(stats).to.deep.equal({
+        processedCheckoutOrders: 1,
+        createdTemplateOrders: 1,
+        failedCheckoutOrders: 0,
+        duplicatedTemplateOrderCreation: 0,
+      })
+      expect(addPaymentToTemplateOrder.isDone()).to.be.true
+      expect(setIsSubscriptionProcessed.isDone()).to.be.true
+      expect(updateLastStartTimestamp.isDone()).to.be.true
+    })
+
+    it('when error is 500 error, retry and fail the job when retry limit is fetched', async () => {
+      _mockCommonRequestsAndResponse()
+
+      const createNewOrder = nock(CTP_API_URL)
+        .post(`/${PROJECT_KEY}/orders/import`)
+        .times(10)
+        .reply(500, {
+          message:
+            '"Client network socket disconnected before secure TLS connection was established"',
+        })
+
+      const setIsSubscriptionProcessed = nock(CTP_API_URL)
+        .post(
+          `/${PROJECT_KEY}/orders/${CHECKOUT_ORDER_ID}`,
+          (body) =>
+            JSON.stringify(body) ===
+            JSON.stringify({
+              actions: [
+                {
+                  action: 'setCustomField',
+                  name: 'isSubscriptionProcessed',
+                  value: true,
+                },
+              ],
+              version: 3,
+            })
+        )
+        .reply(200)
+
+      const updateLastStartTimestamp = nock(CTP_API_URL)
+        .post(
+          `/${PROJECT_KEY}/custom-objects`,
+          (body) =>
+            body.container === 'commercetools-subscriptions' &&
+            body.key === 'subscriptions-lastStartTimestamp'
+        )
+        .reply(200)
+
+      const stats = await createTemplateOrders({
+        ctpClient,
+        apiRoot,
+        logger,
+        startDate: new Date(),
+      })
+
+      expect(stats).to.deep.equal({
+        processedCheckoutOrders: 1,
+        createdTemplateOrders: 0,
+        failedCheckoutOrders: 1,
+        duplicatedTemplateOrderCreation: 0,
+      })
+      expect(createNewOrder.isDone()).to.be.true
```

Comment:
this test is failing on my local, I think there is no retry in code 🤔

Reply:
Correct, check the description 😉

Reply:
OK do you know when it will be fixed ?